Makefile

```
@@ -735,17 +748,27 @@ init-submodules-e: init-webapps-submodules-e
 	git submodule init e
 	git submodule update
 
+# Update go.mod and vendor files. 
 .PHONY: update-vendor
 update-vendor:
 	# update modules in api/
 	cd api && go mod tidy
 	# update modules in root directory
 	go mod tidy
 	go mod vendor
-	# delete the vendored api package. In its place
-	# create a symlink to the the original api package
-	rm -r vendor/github.com/gravitational/teleport/api
-	ln -s -r $(shell readlink -f api) vendor/github.com/gravitational/teleport
+	$(MAKE) vendor-api
+
+# When teleport is vendoring, Go also vendors the local api sub module. To get
+# around this issue, we replace the vendored api package with a symlink to the
+# local module. The symlink should be in vendor/.../api or vendor/.../api/vX if X >= 2.
+.PHONY: vendor-api
+vendor-api:
+	$(eval MOD_PATH=$(shell head -1 api/go.mod | awk '{print $$2;}'))
+	rm -rf vendor/github.com/gravitational/teleport/api
+	# only create api directory if we are linking the api as vX
+	if grep -q "/v[0-9]\+" <<< "$(MOD_PATH)"; \
+		then mkdir -p vendor/github.com/gravitational/teleport/api; fi;\
+	ln -s -r $(shell readlink -f api) vendor/$(MOD_PATH)
```

I may be missing something, wouldn't this better express what's going on here? 
```suggestion
	mkdir -p vendor/$(shell dirname $(MOD_PATH))
	ln -rs $(shell readlink -f api) vendor/$(MOD_PATH)
```